### PR TITLE
fix(ActivityCenter): Simplify and fix AC notifications read/unread states

### PR DESCRIPTION
**Pull request by MishkaRogachev**
Close https://github.com/status-im/status-desktop/issues/12619

### What does the PR do

- [x] Simplify read and unread states for AC notifications making them match status-go state
- [x] Remove accept and dismiss actions (completely handled in status-go)
- [x] `Mark all read` marks ALL notifications as read (it's how it's implemented in status-go)
- [x] Fix wrong disabled state for  `Mark all read`

### Affected areas

Activity center

### Screenshot of functionality (including design for comparison)

https://github.com/status-im/status-desktop/assets/2522130/10113333-e5b1-457a-8436-a1291be50dc6

### CODE REVIEW

**Review by jrainville — COMMENTED** [review]
Nice job. It's way cleaner!

File: src/app_service/service/community_tokens/service.nim [hunk 1]
```diff
@@ -316,12 +316,8 @@ QtObject:
       self.acService.parseActivityCenterResponse(response)
       let notificationToSetRead = self.acService.getNotificationForTypeAndCommunityId(notification.ActivityCenterNotificationType.OwnerTokenReceived, contractDetails.communityId)
       if notificationToSetRead != nil:
-        let markAsReadProps = MarkAsReadNotificationProperties(
-          notificationIds: @[notificationToSetRead.id],
-          communityId: contractDetails.communityId,
-          notificationTypes: @[notification.ActivityCenterNotificationType.OwnerTokenReceived]
-        )
-        discard self.acService.markActivityCenterNotificationRead(notificationToSetRead.id, markAsReadProps)
+        # NOTE: can this be implemented on the status-go side?
```
[the remark this answers is not in the document]
Comment (MishkaRogachev) [reply src/app_service/service/community_tokens/service.nim]: comment removed

File: src/app/modules/main/activity_center/controller.nim [hunk 2]
```diff
@@ -54,36 +54,32 @@ proc init*(self: Controller) =
   self.events.on(activity_center_service.SIGNAL_ACTIVITY_CENTER_NOTIFICATIONS_LOADED) do(e: Args):
     let args = ActivityCenterNotificationsArgs(e)
     self.delegate.addActivityCenterNotifications(args.activityCenterNotifications)
+    self.delegate.hasUnseenActivityCenterNotificationsChanged()
```
[the remark this answers is not in the document]
Comment (MishkaRogachev) [reply src/app/modules/main/activity_center/controller.nim]: fixed (others also)

**Review by jrainville — APPROVED** [review]

File: src/app_service/service/activity_center/service.nim [hunk 3]
```diff
@@ -228,47 +203,28 @@ QtObject:
...
-      markAsReadProps: MarkAsReadNotificationProperties
-      ): string =
+  proc markActivityCenterNotificationRead*(self: Service, notificationId: string) =
     try:
-      discard backend.markActivityCenterNotificationsRead(@[notificationId])
-      self.events.emit(SIGNAL_MARK_NOTIFICATIONS_AS_READ, markAsReadProps)
-      self.events.emit(SIGNAL_ACTIVITY_CENTER_NOTIFICATIONS_COUNT_MAY_HAVE_CHANGED, Args())
+      let notificationIds = @[notificationId]
+      discard backend.markActivityCenterNotificationsRead(notificationIds)
+      self.events.emit(SIGNAL_ACTIVITY_CENTER_MARK_NOTIFICATIONS_AS_READ, ActivityCenterNotificationIdsArgs(notificationIds: notificationIds))
```
Comment (igor-sirotin) [inline src/app_service/service/activity_center/service.nim:210]: What exactly are the cases when we still need to emit `SIGNAL_ACTIVITY_CENTER_NOTIFICATIONS_COUNT_MAY_HAVE_CHANGED`? You removed it in a few places, but it's still there sometimes.

**Review by igor-sirotin — APPROVED** [review]
Looks much cleaner indeed! 👍
Always happy to see more code removed than added 😄

File: src/app_service/service/activity_center/service.nim (hunk 3 above)
Comment (MishkaRogachev) [reply src/app_service/service/activity_center/service.nim:210]: There were extra calls, but its still needed

**Pull request merged**